openeo-geotrellis/src/main/scala/org/openeo/geotrellis/netcdf/NetCDFRDDWriter.scala

```diff
@@ -801,6 +888,120 @@ object NetCDFRDDWriter {
+      bandNames.forEach(name => {
+        val rasterBands = new java.util.HashMap[String,Any]()
+        rasterBands.put("name", name)
+        map.add(rasterBands)
+      })
+      map
+    }
+    assetMetadata.put("bands", bands)
+    assetMetadata.put("proj:bbox", Array(metadata.extent.xmin, metadata.extent.ymin, metadata.extent.xmax, metadata.extent.ymax))
+    assetMetadata.put("proj:epsg", metadata.crs.epsgCode.getOrElse(metadata.crs.proj4jCrs.getName))
+    assetMetadata.put("proj:shape", Array(metadata.layout.tileRows, metadata.layout.tileCols))
+    assetMetadata
+  }
+
+  private def setupAssetMetadata[K: SpatialComponent : Boundable : ClassTag](metadata: TileLayerMetadata[K], rasters: Seq[Raster[MultibandTile]], dates: Seq[ZonedDateTime], bandNames: ArrayList[String], addBandsStats: Boolean): util.Map[String, Any] = {
+    val assetMetadata = new util.HashMap[String,Any]()
+    if (dates != null) {
+      assetMetadata.put("time", Map("type" -> "temporal", "extent" -> Array(dates.head, dates.last), "values" -> dates.toArray))
+    } else new util.HashMap[String,Any]()
+    val bands = if (addBandsStats) {
+      bandsStatistics(rasters, bandNames)
+    } else {
+      val map = new java.util.ArrayList[java.util.HashMap[String,Any]]()
+      bandNames.forEach(name => {
+        val rasterBands = new java.util.HashMap[String,Any]()
+        rasterBands.put("name", name)
+        map.add(rasterBands)
+      })
+      map
+    }
+    assetMetadata.put("bands", bands)
+    assetMetadata.put("proj:bbox",Array(metadata.extent.xmin, metadata.extent.ymin, metadata.extent.xmax, metadata.extent.ymax))
+    assetMetadata.put("proj:epsg", metadata.crs.epsgCode.getOrElse(metadata.crs.proj4jCrs.getName))
+    assetMetadata.put("proj:shape", Array(metadata.layout.tileRows, metadata.layout.tileCols))
+    assetMetadata
+  }
+
+  private def bandsStatistics(tile:Tile, histograms:collection.mutable.Map[String,(StreamingHistogram,Int)], bandName:String):collection.mutable.Map[String,(StreamingHistogram,Int)] = {
+    val result = if (histograms.contains(bandName)) {
+      val (histogram,size) = histograms(bandName)
+      (histogram.merge(tile.histogramDouble()),size+tile.size)
+    } else (StreamingHistogram(tile.histogramDouble()),tile.size)
+    histograms.update(bandName,result)
+    histograms
+  }
+
+  private def bandsStatistics(rasters:Seq[Raster[MultibandTile]], bandNames: ArrayList[String]): java.util.ArrayList[java.util.HashMap[String,Any]] = {
+    val stats = new java.util.ArrayList[java.util.HashMap[String,Any]]()
+    for (bandId <- 0 until bandNames.size()){
+      val histogramsAndSizes = rasters.map(raster => {
+        (raster.tile.band(bandId).histogramDouble(),raster.tile.size)
+      })
+      val (result,size )= histogramsAndSizes.foldLeft((StreamingHistogram(),0))((tempResult,histogramAndSize) => {
+        (tempResult._1.merge(histogramAndSize._1),tempResult._2+histogramAndSize._2)
+      })
```

Comment:
```suggestion
      val (result, size )= histogramsAndSizes.foldLeft((StreamingHistogram(), 0)) { case ((accHistogram, accSize), (histogram, size)) =>
        (accHistogram merge histogram, accSize + size)
      }
```

Pattern matching over `_1` and `_2` for clarity.